pkg/controllers/raycluster_webhook.go

```diff
@@ -100,24 +100,25 @@ func (w *rayClusterWebhook) Default(ctx context.Context, obj runtime.Object) err
 		}
 
 		// WorkerGroupSpec
-
-		// Append the list of environment variables for the worker container
-		for _, envVar := range envVarList() {
-			rayCluster.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].Env = upsert(rayCluster.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].Env, envVar, withEnvVarName(envVar.Name))
-		}
-
-		// Append the CA volumes
-		for _, caVol := range caVolumes(rayCluster) {
-			rayCluster.Spec.WorkerGroupSpecs[0].Template.Spec.Volumes = upsert(rayCluster.Spec.WorkerGroupSpecs[0].Template.Spec.Volumes, caVol, withVolumeName(caVol.Name))
-		}
-
-		// Append the certificate volume mounts
-		for _, mount := range certVolumeMounts() {
-			rayCluster.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].VolumeMounts = upsert(rayCluster.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].VolumeMounts, mount, byVolumeMountName)
+		if len(rayCluster.Spec.WorkerGroupSpecs) != 0 {
+			// Append the list of environment variables for the worker container
+			for _, envVar := range envVarList() {
+				rayCluster.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].Env = upsert(rayCluster.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].Env, envVar, withEnvVarName(envVar.Name))
```

Comment:
RayCluster can contain more WorkerGroupSpecs. Would it have sense to rather loop all existing WorkerGroupSpecs to verify/modify all of them?

Reply:
That makes complete sense, great catch thanks Karel!